docs/spec/generics.rst

```diff
@@ -1761,6 +1763,348 @@ overloads for each possible rank is, of course, a rather cumbersome
+   reveal_type(OneDefault[float]()) # type is OneDefault[float, bool]
+
+
+   class AllTheDefaults(Generic[T1, T2, DefaultStrT, DefaultIntT, DefaultBoolT]): ...
+
+   reveal_type(AllTheDefaults)                  # type is type[AllTheDefaults[T1, T2, DefaultStrT = str, DefaultIntT = int, DefaultBoolT = bool]]
+   reveal_type(AllTheDefaults[int, complex]())  # type is AllTheDefaults[int, complex, str, int, bool]
+   AllTheDefaults[int]  # Invalid: expected 2 arguments to AllTheDefaults
+   (
+       AllTheDefaults[int, complex] ==
+       AllTheDefaults[int, complex, str] ==
+       AllTheDefaults[int, complex, str, int] ==
+       AllTheDefaults[int, complex, str, int, bool]
+   )  # All valid
+
+With the new Python 3.12 syntax for generics (introduced by :pep:`695`), this can
+be enforced at compile time::
+
+   type Alias[DefaultT = int, T] = tuple[DefaultT, T]  # SyntaxError: non-default TypeVars cannot follow ones with defaults
+
+   def generic_func[DefaultT = int, T](x: DefaultT, y: T) -> None: ...  # SyntaxError: non-default TypeVars cannot follow ones with defaults
+
+   class GenericClass[DefaultT = int, T]: ...  # SyntaxError: non-default TypeVars cannot follow ones with defaults
+
+``ParamSpec`` Defaults
+''''''''''''''''''''''
+
+``ParamSpec`` defaults are defined using the same syntax as
+``TypeVar`` \ s but use a ``list`` of types or an ellipsis
+literal "``...``" or another in-scope ``ParamSpec`` (see `Scoping Rules`_).
+
+::
+
+   DefaultP = ParamSpec("DefaultP", default=[str, int])
+
+   class Foo(Generic[DefaultP]): ...
+
+   reveal_type(Foo)                  # type is type[Foo[DefaultP = [str, int]]]
+   reveal_type(Foo())                # type is Foo[[str, int]]
+   reveal_type(Foo[[bool, bool]]())  # type is Foo[[bool, bool]]
+
+``TypeVarTuple`` Defaults
+'''''''''''''''''''''''''
+
+``TypeVarTuple`` defaults are defined using the same syntax as
+``TypeVar`` \ s but use an unpacked tuple of types instead of a single type
+or another in-scope ``TypeVarTuple`` (see `Scoping Rules`_).
+
+::
+
+   DefaultTs = TypeVarTuple("DefaultTs", default=Unpack[tuple[str, int]])
+
+   class Foo(Generic[*DefaultTs]): ...
+
+   reveal_type(Foo)               # type is type[Foo[DefaultTs = *tuple[str, int]]]
+   reveal_type(Foo())             # type is Foo[str, int]
+   reveal_type(Foo[int, bool]())  # type is Foo[int, bool]
+
+Using Another Type Parameter as ``default``
+''''''''''''''''''''''''''''''''''''''''''''
+
+This allows for a value to be used again when the type parameter to a
+generic is missing but another type parameter is specified.
+
+To use another type parameter as a default the ``default`` and the
+type parameter must be the same type (a ``TypeVar``'s default must be
+a ``TypeVar``, etc.).
+
+`This could be used on builtins.slice <https://github.com/python/typing/issues/159>`__
+where the ``start`` parameter should default to ``int``, ``stop``
```

Comment:
```suggestion
For instance, this could be used to type something like builtins.slice,
where the ``start`` parameter should default to ``int``, ``stop``
```
nit: original language was more suited for proposal, sounds dated if typeshed starts using 696 on slice